public/js/leaderboard.js

```diff
@@ -1,19 +1,99 @@
 /* set home as the active navbar element */
 $('#nav-leaderboard').addClass('active');
 
+var boardType = leaderboardTypes.OVERALL;
+var leaderboardRow;
+var leaderboardTable;
+var studentLeaderList;
+var podiumScore;
+
 /* Get HTML for the complete leaderboard table from the server and display it. */
 var fetchLeaderboard = function() {
+    
     $.ajax({
         type: 'GET',
         url: '/leaderboard-table',
         data: {
-            fullTable: true,
-            longTable: true
+            smallBoard: false
         },
         success: function(data) {
-            $('.leaderboard-content').html(data);
+            setPodiumImages();
+            leaderboardTable = $(data.leaderboardTableHTML);
+            leaderboardRow = $(data.leaderboardRowHTML);
+            studentLeaderList = data.leader;
+            $('.leaderboard-content').html(leaderboardTable);
+            displayNewLeaderboard(boardType);
         }
     });
 }
 
 fetchLeaderboard();
+
+var displayLeaderboard = function(studentLeaderList) {
+    var prevRank;
+    var rank;
+
+    $('#leaderboardBody').html('');
+    $('#criteriaName').html(boardType.displayName);
+    podiumScore = studentLeaderList[0][boardType.name];
+    $('#topScore').html(podiumScore + ((boardType === leaderboardTypes.ACCURACY) ? '%' : ''));
+    studentLeaderList.forEach((studentObject, index) => {
+        // Students with the same number of points get the same rank
+        if (index === 0) {
+            leaderboardRow.find('#rank').html(index + 1);
+            rank = 1;
+            prevRank = rank;
+        } else {
+            if (studentLeaderList[index - 1][boardType.name] === studentObject[boardType.name]) {
+                rank = prevRank;
+                leaderboardRow.find('#rank').html(rank);
+            } else {
+                rank = index + 1;
+                prevRank = rank;
+                leaderboardRow.find('#rank').html(rank);
+            }
+        }
+        // This give colour to rows where the student's rank is in the top 3       
+        leaderboardRow.attr('class', `rank-${rank <= 3 ? rank : 'default'}`);
+        leaderboardRow.find('#displayName').html(studentObject.displayName);
+        leaderboardRow.find('#criteria').html(studentObject[boardType.name] + 
```

Comment:
this should be a separate block, for adding the`%` sign

Reply:
Fixed